langkit/templates/iterator_types_ada.mako

```diff
@@ -5,7 +5,10 @@
 
   function Next
     (I : ${cls.api_name};
-     E : out ${cls.element_type.api_name}) return Boolean;
+     E : out ${"{}{}".format(
+         cls.element_type.api_name,
+         "'Class" if cls.element_type.is_entity_type else ''
+     )}) return Boolean;
```

Comment:
This looks wrong to me: assuming that B derives from A, if we have an iterator on A, we don’t want to allow users to pass a B actual (which matches A'Class).